src/Corax/IndexEntry/IndexEntryWriter.cs

```diff
@@ -91,236 +117,254 @@ public void WriteRaw(int field, ReadOnlySpan<byte> binaryValue)
         //STRUCT
         //<type><size_of_binary><binary>
         Debug.Assert(field < _knownFields.Count);
-        Debug.Assert(_knownFieldsLocations[field] == Invalid);
+        Debug.Assert(KnownFieldsLocations[field] == Invalid);
+
+        if (FreeSpace < binaryValue.Length + sizeof(long))
+            UnlikelyGrowAuxiliaryBuffer();
 
         int dataLocation = _dataIndex;
 
         // Write known field.
-        ref int fieldLocation = ref _knownFieldsLocations[field];
+        ref int fieldLocation = ref KnownFieldsLocations[field];
         fieldLocation = dataLocation | Constants.IndexWriter.IntKnownFieldMask;
 
-        // Write the list metadata information. 
-        fixed (byte* bufferPtr = _buffer)
-        {
-            ref var indexEntryField = ref Unsafe.AsRef<IndexEntryFieldType>(bufferPtr + dataLocation);
-            indexEntryField = IndexEntryFieldType.Raw;
-            dataLocation += sizeof(IndexEntryFieldType);
+        var buffer = Buffer;
 
-            if (binaryValue.Length == 0)
-            {
-                // Signal that we will have to deal with the empties.
-                indexEntryField |= IndexEntryFieldType.Empty;
-            }
-            else
-            {
-                dataLocation += VariableSizeEncoding.Write(_buffer, binaryValue.Length, dataLocation);
+        // Write the list metadata information. 
+        ref var indexEntryField = ref Unsafe.AsRef<IndexEntryFieldType>(Unsafe.AsPointer(ref buffer[dataLocation]));
```

Comment:
Avoid this.

Reply:
@redknightlois I see in the commit https://github.com/ravendb/ravendb/pull/14514/commits/fc3340b78e2b312b696cdb37c7f06963884e1199 you addressed some comments to avoid dangling pointers but not this one. Shouldn't the result of `Unsafe.AsPointer()` be wrapped by `fixed()`?

Reply:
In the case of `IndexEntryWriter` we own the memory and we know it comes from unmanaged; so we can do that without any required usage of `fixed`. We use `Span<byte>` for convenience and bound check semantics; but we could have very well used the pointers directly.